solution.js

```diff
@@ -0,0 +1,60 @@
+const array = [
+    { id: 1, name: "Bilal", city: "Lahore" },
+    { id: 1, name: "Bilal", city: "Lahore" },
+    { id: 3, name: "Hafsa", city: "Karachi" },
+    { id: 4, name: "Rehan", city: "Lahore" },
+    { id: 5, name: "Saqib", city: "Karachi" },
+    { id: 6, name: "Farhan", city: "Islamabad" }
+];
+
+function GroupByUsingReduce(array, attribute) {
+    let result = array.reduce((acc, user) => {
```

Comment:
try to use const where possible like 
`const groupedData` but i believe we can directly return we don't even need to declare a constant here